### PR TITLE
Add pagination for categories and phrases

### PR DESCRIPTION
**Pull request by nicholas-cook**
- Adds pagination functionality for the categories and phrases on the presets screen
- Uses ViewPager2s and fragments to accomplish pagination
- NOTE - Pagination only works for swiping and not head tracking

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt [hunk 1]
```diff
@@ -53,4 +112,39 @@ class PresetsFragment : BaseFragment() {
...
+    }
+
+    inner class PhrasesPagerAdapter(fm: FragmentManager, private val phrases: List<String>) :
+        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
+
+        override fun getItemCount(): Int {
+            return ceil(phrases.size / MAX_PHRASES.toDouble()).toInt()
+        }
+
+        override fun createFragment(position: Int): Fragment {
```
Comment (CarolineLaw) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt]: we should have infinite scrolling eventually :]

File: app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        CATEGORY_CONVERSATION,
+        CATEGORY_ENVIRONMENT,
+        CATEGORY_FEELINGS,
+        CATEGORY_QUESTIONS,
+        CATEGORY_TIME,
+        CATEGORY_PEOPLE,
+        CATEGORY_NUMBERS
+    )
+
+    private val generalPhrases = listOf(
```
Comment (jequeen) [inline app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt]: In the future we could put our categories and phrase lists into JSON and parse those maybe.

File: app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt (hunk 1 above)
Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt]: That's coming in the next PR!

File: app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt (hunk 2 above)
Comment (nicholas-cook) [reply app/src/main/java/com/willowtree/vocable/presets/PresetsViewModel.kt]: Yeah I think that will definitely clean up this view model 👍

**Pull request merged**